### PR TITLE
Fix release display and correct paging

### LINKED ISSUE #1863: Release page don't paginate
## Env

- Gitea version (or commit ref): 009a0739906566abd511d17d53b8282696e9e99e
- Git version:
- Operating system:
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes (https://try.gitea.io/sapk/test-release/releases && https://try.gitea.io/sapk/test-release/releases?page=2)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

The page list is not shown at bottom of page

## Screenshots

image

### PR DESCRIPTION
**Pull request by lafriks**
* Fixes displaying releases with paging when there are drafts and user has no rights to see drafts
* Fix so that drafts can be seen not only by Owner but anyone with write rights to repository
* Add integration tests to create new releases and paging

Fixes #1863

### CODE REVIEW

**Comment by lafriks** [comment]
Continuation to #2035

File: models/release.go [hunk 1]
```diff
@@ -233,40 +233,44 @@ func GetReleaseByID(id int64) (*Release, error) {
...
-	return rels, err
+// FindReleasesOptions describes the conditions to Find releases
+type FindReleasesOptions struct {
+	IncludeDrafts bool
+	TagNames      []string
 }
 
-// GetReleaseCountByRepoID returns the count of releases of repository
-func GetReleaseCountByRepoID(repoID int64, includeDrafts bool) (int64, error) {
+func (opts *FindReleasesOptions) toConds(repoID int64) builder.Cond {
```
Comment (lunny) [inline models/release.go:242]: Why not let `repoID` as a field of FindReleasesOptions?

File: models/release.go [hunk 2]
```diff
@@ -233,40 +233,44 @@ func GetReleaseByID(id int64) (*Release, error) {
...
+	if len(opts.TagNames) > 0 {
+		cond = cond.And(builder.In("tag_name", opts.TagNames))
+	}
+	return cond
 }
 
-// GetReleasesByRepoIDAndNames returns a list of releases of repository according repoID and tagNames.
-func GetReleasesByRepoIDAndNames(repoID int64, tagNames []string) (rels []*Release, err error) {
+// GetReleasesByRepoID returns a list of releases of repository.
+func GetReleasesByRepoID(repoID int64, opts FindReleasesOptions, page, pageSize int) (rels []*Release, err error) {
```
Comment (lunny) [inline models/release.go:256]: As above and page and pageSize is also could be fields of `FindReleasesOptions`

File: models/release.go (hunk 1 above)
Comment (lafriks) [reply models/release.go:242]: At first I did that but than it looked strange as func name is `GetReleasesByRepoID`. And also `repoID` is mandatory condition

Comment (lunny) [reply models/release.go:242]: I think `FindReleases` and `CountReleases` maybe a new name?

File: models/release.go (hunk 2 above)
Comment (lafriks) [reply models/release.go:256]: Because it's not really related to conditions and toConds can't use them anyway

File: models/release.go (hunk 1 above)
Comment (lafriks) [reply models/release.go:242]: But repoID is still allways required field and that will not easy be understandable by function signature

Comment (lunny) [reply models/release.go:242]: seems right.

**Comment by ethantkoenig** [comment]
LGTM

**Comment by lunny** [comment]
LGTM

**Pull request merged**